src/vs/workbench/browser/web.main.ts

```diff
@@ -92,6 +92,10 @@ class BrowserMain extends Disposable {
 		// Layout
 		this._register(addDisposableListener(window, EventType.RESIZE, () => workbench.layout()));
 
+		if ((<any>window).visualViewport) {
```

Comment:
Are we getting 2 layout calls then if the window size changes to a small width?

Reply:
Good catch! It did trigger two layouts, which should be avoided.

Considering there is no builtin way of merging these two events and desktop browsers support visual viewport as well, now I narrow it to use visual viewport resize instead of window resize only on iOS13/iPadOS, based on what I tested.